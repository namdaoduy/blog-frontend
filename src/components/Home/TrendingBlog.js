import Badge from '@material-ui/core/Badge';
import ButtonBase from '@material-ui/core/ButtonBase';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import React, { Component } from 'react';
import Moment from 'react-moment';
import history from '../../utils/history';

export default class TrendingBlog extends Component {
  handleClick = () => {
    const { blog } = this.props;
    history.push(`/blog/${blog.id || 0}`);
  }

  render() {
    const { blog, rank } = this.props;
    return (
      <Grid item xs={12}>
        <ButtonBase className="btn-base" onClick={this.handleClick}>
          <Badge badgeContent={rank} color="secondary" className="badge left">
            <Paper className="home-new-blog">
              <Typography variant="subtitle1" className="serif-2">
                {blog.title || 'Blog Title'}
              </Typography>
              <Typography noWrap>
                {blog.body || 'Blog body'}
              </Typography>
              <Typography variant="subtitle2">
                {blog.author || 'Blog Author'}
              </Typography>
              <Typography variant="caption">
                <Moment fromNow>{blog.created_at || 1545813100264}</Moment>
                <span className="dot-divider" />
                {blog.timeRead || `${10} min read`}
              </Typography>
            </Paper>
          </Badge>
        </ButtonBase>
      </Grid>
    );
  }
}