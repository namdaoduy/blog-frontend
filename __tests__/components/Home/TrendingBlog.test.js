import React from 'react';
import { shallow } from 'enzyme';
import history from 'utils/history';
import TrendingBlog from 'components/Home/TrendingBlog';

describe('components/Home/TrendingBlog.js', () => {
  let props;
  let wrapper;

  const setup = () => {
    wrapper = shallow(<TrendingBlog {...props} />);
  };

  beforeEach(() => {
    props = {
      blog: {
        id: 1,
        title: 'Mock title',
        body: 'Mock body',
        author: 'Mock Author',
        created_at: 123456789,
      },
    };
  });

  describe('Render correctly', () => {
    it('When blog is valid', () => {
      setup();
      expect(wrapper).toMatchSnapshot();
    });

    it('When blog is invalid', () => {
      props.blog = {};
      setup();
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Change route correctly', () => {
    setup();
    wrapper.instance().handleClick();
    expect(history.push).toHaveBeenCalledWith(`/blog/${props.blog.id}`);
  });
});
