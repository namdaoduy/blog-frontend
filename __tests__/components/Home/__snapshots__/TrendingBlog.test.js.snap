// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Home/TrendingBlog.js Render correctly When blog is invalid 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(ButtonBase)
    className="btn-base"
    onClick={[Function]}
  >
    <WithStyles(Badge)
      className="badge left"
      color="secondary"
    >
      <WithStyles(Paper)
        className="home-new-blog"
      >
        <WithStyles(Typography)
          className="serif-2"
          variant="subtitle1"
        >
          Blog Title
        </WithStyles(Typography)>
        <WithStyles(Typography)
          noWrap={true}
        >
          Blog body
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="subtitle2"
        >
          Blog Author
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="caption"
        >
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            fromNow={true}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            1545813100264
          </t>
          <span
            className="dot-divider"
          />
          10 min read
        </WithStyles(Typography)>
      </WithStyles(Paper)>
    </WithStyles(Badge)>
  </WithStyles(ButtonBase)>
</WithStyles(Grid)>
`;

exports[`components/Home/TrendingBlog.js Render correctly When blog is valid 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(ButtonBase)
    className="btn-base"
    onClick={[Function]}
  >
    <WithStyles(Badge)
      className="badge left"
      color="secondary"
    >
      <WithStyles(Paper)
        className="home-new-blog"
      >
        <WithStyles(Typography)
          className="serif-2"
          variant="subtitle1"
        >
          Mock title
        </WithStyles(Typography)>
        <WithStyles(Typography)
          noWrap={true}
        >
          Mock body
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="subtitle2"
        >
          Mock Author
        </WithStyles(Typography)>
        <WithStyles(Typography)
          variant="caption"
        >
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            fromNow={true}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            123456789
          </t>
          <span
            className="dot-divider"
          />
          10 min read
        </WithStyles(Typography)>
      </WithStyles(Paper)>
    </WithStyles(Badge)>
  </WithStyles(ButtonBase)>
</WithStyles(Grid)>
`;
